crates/optimism/rpc/src/engine.rs

```diff
@@ -206,13 +208,18 @@ impl<Provider, EngineT, Pool, Validator, ChainSpec> OpEngineApiServer<EngineT>
     for OpEngineApi<Provider, EngineT, Pool, Validator, ChainSpec>
 where
     Provider: HeaderProvider + BlockReader + StateProviderFactory + 'static,
-    EngineT: EngineTypes<ExecutionData = ExecutionData>,
+    EngineT: EngineTypes<ExecutionData = OpExecutionData>,
     Pool: TransactionPool + 'static,
     Validator: EngineValidator<EngineT>,
     ChainSpec: EthereumHardforks + Send + Sync + 'static,
 {
     async fn new_payload_v2(&self, payload: ExecutionPayloadInputV2) -> RpcResult<PayloadStatus> {
-        EngineApiServer::new_payload_v2(&self.inner, payload).await
+        trace!(target: "rpc::engine", "Serving engine_newPayloadV2");
+        let payload = OpExecutionData {
+            payload: payload.into_payload(),
+            sidecar: ExecutionPayloadSidecar::none(),
+        };
```

Comment:
@emhane the only thing missing here are these conversions and it looks like we're missing some from impls on op-alloy

other than that all of that should work now